### PR TITLE
Documenation

### PR DESCRIPTION
**Pull request by Horaceshi624**
Docamentation for pysqldb3

### CODE REVIEW

File: pysqldb3/README.md [hunk 1]
```diff
@@ -1 +1,88 @@
...
+1. [`check_conn`](#check_conn): Checks and reconnects to connection if not currently connected.
+
+### DbConnect
+**`pysqldb3.DbConnect.(quiet=False)`**
+Creates database connection instance.  
+###### Parameters:
+ - **`quiet`: bool, default False**: When true, does not print database connection information.
+ 
+ 
+**Sample**
```
Comment (shostetter) [inline pysqldb3/README.md:34]: this sample doesnt show the `db.connect`, its showing `db.DbConnect` call

File: pysqldb3/README.md [hunk 2]
```diff
@@ -1 +1,88 @@
...
+Checks and reconnects to connection if not currently connected.
+
+###### Parameters: 
+- None
+
+**Sample**
+```
+>>> from ris import pysqldb
+>>> db = pysqldb.DbConnect(type='pg', server=server_address, database='ris', user='shostetter', password='*******')
+>>> db.check_conn()
```
Comment (shostetter) [inline pysqldb3/README.md:84]: this function doesnt return anything, so maybe the sample inst needed. but you should add that nothing is returned (look at how that is noted elsewhere in the doc to be consistent).

**Review by shostetter — CHANGES_REQUESTED** [review]
Looks ok, made some comments for minor changes

File: pysqldb3/README.md (hunk 2 above)
Comment (Horaceshi624) [reply pysqldb3/README.md:84]: ok. in the doc of RIS, for check_conn there is also a sample without output. but there are some other functions in doc of RIS neither have output or sample.

File: pysqldb3/README.md (hunk 1 above)
Comment (Horaceshi624) [reply pysqldb3/README.md:34]: ok, will show db.connect

File: pysqldb3/README.md [hunk 3]
```diff
@@ -1 +1,1027 @@
...
+<br>
+
+#### table_to_shp
+**`DbConnect.table_to_shp(table, schema=None, strict=True, path=None, shp_name=None, cmd=None,
+                     gdal_data_loc=GDAL_DATA_LOC, print_cmd=False)`**
+
+Generates shapefile output from the data returned from a query. 
+
+###### Parameters:
+ - **`table` str**: SQL query as string type; the query should ultimatley return data (ie. include a `select` statement with polygon geometry or must contain appropriate join attrbute)
```
Comment (shostetter) [inline pysqldb3/README.md:846]: this looks like its from query_to_shp not table_to_shp

File: pysqldb3/README.md [hunk 4]
```diff
@@ -1 +1,1027 @@
...
+                               private=False, temp=True, fc_encoding=None, print_cmd=False,
+                               days=7)`**
+
+Imports feature class from ESRI file geodatabase, uses GDAL to generate the table.
+
+###### Parameters:
+ - **`path` str**: File path of the geodatabase
+ - **`table` str**: Table name to be used in the database
+ - **`schema` str, default None**:  Database schema to use for destination in database (defaults database object's default schema)
+ - **`shp_name` str, default None**: Output filename to be used for shapefile
```
Comment (shostetter) [inline pysqldb3/README.md:993]: this doesnt make sense, since there is no output shp. This must be an input name

**Review by shostetter — CHANGES_REQUESTED** [review]
Hao this looks good. I made some comments for some relatively small changes. Once you have fix those let me know and we can do a final review and merge it in.

File: pysqldb3/README.md [hunk 5]
```diff
@@ -1 +1,1027 @@
...
+['pg_catalog', 'information_schema', 'public', 'topology', 'working', 'staging', 'archive', 'visionzero']
+```
+[Back to Table of Contents](#pysqldb3-public-functions)
+<br>
+
+### query
+**`DbConnect.query(query, strict=True, permission=True, temp=True, timeme=True, no_comment=False, comment='',
+              lock_table=None, return_df=False, days=7, internal=False)`**
+
+Runs query from input SQL string, calls Query object.
```
[the remark this answers is not in the document]
Comment (Horaceshi624) [reply pysqldb3/README.md:262]: ok. Should I add this after line 260? "all queries run are logged in the DbConnection.queries attribute, tables created in any query call are logged in DbConnection.tables_created, and tables dropped are logged in DbConnection.tables_dropped"

File: pysqldb3/README.md (hunk 3 above)
Comment (Horaceshi624) [reply pysqldb3/README.md:848]: ah ok, I saw in ris README this issue may also exists.

File: pysqldb3/README.md [hunk 6]
```diff
@@ -299,7 +297,7 @@ Runs query from input SQL string, calls Query object.
  - **`timeme` bool, default True**: If False overrides default behavior that automatically prints query durration time
  - **`no_comment` bool, default False**: If True overrides default behavior to automatically generate a comment on any tables created in query (Postgres only)
  - **`comment` str, default ''**: If provided, appends to automatic table generation comment
- - **`lock_table` str, default None**: ??? Table schema and name to be locked in formate `'schema.table'`
+ - **`lock_table` str, default None**: ??? Table schema and name to be locked in format `'schema.table'`
```
Comment (shostetter) [inline pysqldb3/README.md:300]: A little more detail here would be helpful. If you follow this param through to the query class it is used for creating exclusive locks on the table. See line 106 in query.py

File: pysqldb3/README.md (hunk 3 above)
Comment (shostetter) [reply pysqldb3/README.md:877]: ah yeah ok we should fix mistakes like this as we find them

File: pysqldb3/README.md [hunk 7]
```diff
@@ -1023,7 +1021,7 @@ Imports feature class from ESRI file geodatabase, uses GDAL to generate the tabl
  - **`path` str**: File path of the geodatabase
  - **`table` str**: Table name to be used in the database
  - **`schema` str, default None**:  Database schema to use for destination in database (defaults database object's default schema)
- - **`shp_name` str, default None**: Output filename to be used for shapefile
+ - **`shp_name` str, default None**: Input filename to be used for shapefile
```
Comment (shostetter) [inline pysqldb3/README.md:1024]: this is the input feature_class name

File: pysqldb3/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (shostetter) [reply pysqldb3/README.md:291]: yeah maybe see if you can make the language fit a little better with the document / make sure its clear

**Comment by shostetter** [comment]
thanks @Horaceshi624, i left a few more comments

**Review by shostetter — APPROVED** [review]
Looks good, Can you check that there are no additional public functions that arent documented in this, and if there are add them to the readme?

**Comment by Horaceshi624** [comment]
Hi Seth,  just checked and added 2 functions.

**Review by shostetter — APPROVED** [review]

**Pull request merged**